matlab/JSBSim_SFunction.cpp

```diff
@@ -500,90 +553,52 @@ static void mdlOutputs(SimStruct *S, int_T tid)
    *    integration step.
    */
-  static void mdlUpdate(SimStruct *S, int_T tid)
-  {
-	/* send update inputs to JSBSimInterface, run one cycle, 
-	   retrieve state vector, and update sim state vector 
-	  */
-
-	 JSBSimInterface *JII = (JSBSimInterface *) ssGetPWork(S)[0];   // retrieve C++ object pointers vector
-     
-	 real_T *x2 = ssGetRealDiscStates(S);
-	 //real_T *x = ssGetContStates(S);
-	 InputRealPtrsType uPtrs = ssGetInputPortRealSignalPtrs(S,0); // Input 
-     throttle = *uPtrs[0];
-     aileron = *uPtrs[1];
-     elevator = *uPtrs[2];
-     rudder = *uPtrs[3];
-     mixture = *uPtrs[4];
-     runset = *uPtrs[5];
-     flaps = *uPtrs[6];
-     gear = *uPtrs[7];
-          
-	 //double *derivatives = (double *) ssGetDWork(S,2);
-
-     // These point to the output 
-	 double *inputs   = (double *) ssGetDWork(S,0);
-	 double *states = (double *) ssGetDWork(S,1);
-	 double *controls = (double *) ssGetDWork(S,3);
-	 double *pilot = (double *) ssGetDWork(S,5);
-     
-	 int k;
-     
-	 for (k=0; k < ssGetDWorkWidth(S,0); k++) {
-        inputs[k] = (*uPtrs[k]);
-     }
-	 for (k=0; k < ssGetDWorkWidth(S,1); k++) {
-        states[k] = x2[k];
-     }
-     
-     double ctrl_vec[8] = {throttle, aileron, elevator, rudder, mixture, runset, flaps, gear};
-
-     if(!script_bool){            
-         if(JII->Copy_Controls_To_JSBSim(ctrl_vec)){
-             //mexPrintf("Success to copy controls \n");
-         }    
-     }    
-
-     JII->Update();    
-     
-     JII->Copy_States_From_JSBSim(states); 
-     JII->Copy_Pilot_From_JSBSim(pilot); 
-     JII->Copy_Control_From_JSBSim(controls); 
-     
-	// From previous code, why is this done twice? 
-	for (k=0; k < ssGetDWorkWidth(S,1); k++) {
-        x2[k] = states[k];
-    } 
-	/* for (k=0; k < ssGetDWorkWidth(S,2); k++) {
-        dx[k] = derivatives[k];
-    }*/
-	//UNUSED_ARG(tid);
-  }
-#endif /* MDL_UPDATE */
+static void mdlUpdate(SimStruct *S, int_T tid)
+{
+    /* send update inputs to JSBSimInterface, run one cycle, 
+    retrieve state vector, and update sim state vector 
+    */
 
+    JSBSimInterface* JII = (JSBSimInterface*) ssGetPWork(S)[0];
 
-// NOT USED! 
-#undef MDL_DERIVATIVES  /* Change to #undef to remove function */
-#if defined(MDL_DERIVATIVES)
-  /* Function: mdlDerivatives =================================================
-   * Abstract:
-   *    In this function, you compute the S-function block's derivatives.
-   *    The derivatives are placed in the derivative vector, ssGetdX(S).
-   */
-  static void mdlDerivatives(SimStruct *S)
-  {
-	  real_T *dx = ssGetdX(S);
-	  double *w2 =  (double *) ssGetDWork(S,2);
-	  for (int i = 0; i < ssGetDWorkWidth(S,2); i++)
-	 {
-		dx[i] = w2[i]; // outputs are the flight control outputs 
-	 }
+    InputRealPtrsType ctrlCmdInput = ssGetInputPortRealSignalPtrs(S, 0);
+    double* dWorkCtrlCmdIn = (double*) ssGetDWork(S, 0);
+    double ctrlVec[inputSize];
```

Comment:
Declare arrays with a size that is unknown at compilation is illegal in C++. Better use `vector<>`:
```suggestion
    vector<double> ctrlVec;
    ctrlVec.reserve(inputSize);
```